README.md

```diff
@@ -1,41 +1,123 @@
+To run tests:
+```shell
+open public/buffalo/SpecRunner.html
+```
 
-## Project overview
+To open the app:
+```shell
+open public/index.html
+```
 
-Your app will grab all the headlines from the Guardian newspaper API and display them on a page.  Clicking on a headline will show a summary of the article.
+## Design
 
-### Technologies
+***On Mac***
 
-You'll write a single page web app.  You'll write your code in frontend JavaScript, CSS and HTML.  You won't use Ruby or backend JavaScript.
+![demo](public/images/demo.gif)
 
-**And, as is the theme for this week, you won't use any libraries or frameworks!**
+***On Mobile***
 
-But, feel free to use the test framework you wrote during the week!
+![mobile](public/images/mobile.gif)
 
-### Serving your app
 
-You'll use a static web server (e.g. [http-server](https://www.npmjs.com/package/http-server)) to serve your HTML, CSS and JavaScript files.  You'll send requests to an API to get data from the Guardian and to summarise text.
+### api.js
+- The api.js code makes requests to the Guardian API for the day's politics articles
 
-> The API is hosted on an external server that you don't have to worry about.  You only need a static web server.  That's why this type of architecture is called "serverless".
+```js
+const POLITICS_URL = `https://content.guardianapis.com/search?from-date=${yesterday()}&to-date=${today()}&q=politics&api-key=test&show-fields=all`
 
-## User Stories
+function fetchArticles(url = POLITICS_URL) {
+  fetch(url, {
+    method: 'GET',
+    mode: 'cors',
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  }).then(response => response.json())
+    .then(data => frontPage.articles = data.response.results);
+}
+```
+
+- Functions `yesterday()` and `today()` generate the relevent dates to insert into the API request query string
+
+- When a user clicks on a headline, the app makes a new request to the Aylien API to get a summary of the article
+
+```js
+function fetchSummaries(articleUrl) {
+  const makersUrl = 'http://news-summary-api.herokuapp.com/aylien?apiRequestUrl='
+  const aylienUrl = 'https://api.aylien.com/api/v1/summarize?url='
+  const url = makersUrl + aylienUrl + articleUrl
 
-Some of these stories will need decomposing if they seem too large.
+  fetch(url, {
+    method: 'GET',
+    mode: 'cors',
+    headers: {
+      // ...
+  }
+}
+```
+
+- Use of test and development environment to stub API calls and ensure rate limit isn't breached
+
+```js
+  if (ENVIRONMENT === 'test') {
+    const results = '{"response":{"status":"ok","results":"Test OK"}}'
+    frontPage.articles = results;
+    return results;
+  }
+```
+
+### interface.js
+
+- Templates the article data into index.html from the JSON responses
+
+```js
+function showHeadlinesAndPictures() {
+  setTimeout(() => {
+    let html = ''
+    for (let i = 0; i < frontPage.articles.length; i ++) {
```

Comment:
what's the reason for having a counter here rather than forEach or for x of y?

Reply:
Arbitrary decision to go with a for loop, no reason in particular 🙂